Загрузите набор данных auto-mpg из ~ / наборов данных / auto-mpg / Проверить нулевую гипотезу, отклонять или не отклонять нулевую гипотезу.
Гипотезa:

Independent variable: origin

Dependent variable: mpg

Ho:  Существенных различий в милях среди транспортных средств различного происхождения не существует.

H1:  Существенная разница существует в милях среди транспортных средств различного происхождения.

In [1]:
import numpy as np
import pandas as pd
data=pd.read_csv('d:\\C#\\DA_NEW\\Datasets\\auto-mpg.csv')
data.head()

mpg  cylinders  displacement horsepower  weight  acceleration  model year  \
0  18.0          8         307.0        130    3504          12.0          70   
1  15.0          8         350.0        165    3693          11.5          70   
2  18.0          8         318.0        150    3436          11.0          70   
3  16.0          8         304.0        150    3433          12.0          70   
4  17.0          8         302.0        140    3449          10.5          70   

   origin                   car name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320  
2       1         plymouth satellite  
3       1              amc rebel sst  
4       1                ford torino

In [2]:
k=data['origin'].unique()
k

array([1, 3, 2], dtype=int64)

У нас 3 производителя зкачит k=3(Всего три группы)

In [4]:
k=3
company1=np.array(data.loc[data['origin']==1,'mpg'],dtype=np.int64)
company2=np.array(data.loc[data['origin']==2,'mpg'],dtype=np.int64)
company3=np.array(data.loc[data['origin']==3,'mpg'],dtype=np.int64)
n1=company1.size
print(n1)
n2=company2.size
print(n2)
n3=company3.size
print(n3)
n=n1+n2+n3
print(n)

249
70
79
398


Найдем ср. скорость для каждого произвадителя

In [5]:
company1_mean=np.mean(company1)
print(company1_mean)
company2_mean=np.mean(company2)
print(company2_mean)
company3_mean=np.mean(company3)
print(company3_mean)

19.927710843373493
27.742857142857144
30.240506329113924


Видно, что средние значение в милях разнятся. Установим, что это отличие статистически значимо. Для этого сначала соберем

все значения ростов в один массив:

In [6]:
y_all=np.concatenate([company1,company2,company3])
y_all

array([18, 15, 18, 16, 17, 15, 14, 14, 14, 15, 15, 14, 15, 14, 22, 18, 21,
       21, 10, 10, 11,  9, 28, 25, 19, 16, 17, 19, 18, 14, 14, 14, 14, 12,
       13, 13, 18, 22, 19, 18, 23, 26, 25, 20, 21, 13, 14, 15, 14, 17, 11,
       13, 12, 13, 15, 13, 13, 14, 22, 28, 13, 14, 13, 14, 15, 12, 13, 13,
       14, 13, 12, 13, 18, 16, 18, 18, 23, 11, 12, 13, 12, 18, 21, 19, 21,
       15, 16, 15, 11, 20, 21, 19, 15, 26, 25, 16, 16, 18, 16, 13, 14, 14,
       14, 28, 19, 18, 15, 15, 16, 15, 16, 14, 17, 16, 15, 18, 21, 20, 13,
       23, 20, 23, 18, 19, 25, 26, 17, 16, 15, 14, 22, 22, 24, 22, 29, 24,
       20, 18, 18, 17, 26, 13, 16, 13, 13, 13, 30, 25, 17, 17, 15, 15, 17,
       20, 19, 18, 16, 15, 15, 16, 24, 25, 30, 33, 36, 19, 19, 20, 19, 20,
       20, 25, 20, 19, 20, 20, 18, 18, 19, 17, 18, 17, 30, 30, 23, 23, 21,
       19, 22, 20, 20, 17, 17, 16, 18, 16, 15, 19, 18, 35, 27, 23, 23, 34,
       34, 28, 28, 26, 33, 32, 28, 26, 24, 19, 27, 23, 27, 26, 25, 23, 30,
       39, 34, 34, 29, 22

Найдем среднее скорость по всем значениям:

In [7]:
y_mean=np.mean(y_all)
print(y_mean)

23.349246231155778


Найдем $S^2$ — сумму квадратов отклонений наблюдений от общего среднего:

In [8]:
s2=np.sum((y_all-y_mean)**2)
s2

23860.45477386935

Найдем $S^2_F$ - сумму квадратов отклонений средних групповых значений от общего среднего:

In [9]:
s2_f = ((company1_mean - y_mean)**2) * n1 + ((company2_mean - y_mean)**2) * n2 + ((company3_mean - y_mean)**2) * n3
s2_f

8017.954170370362

Найдем $S^2_{ост}$ — остаточную сумму квадратов отклонений:

In [10]:
s2_residual = np.sum((company1 - company1_mean)**2) + np.sum((company2 - company2_mean)**2) + np.sum((company3 - company3_mean)**2)
s2_residual

15842.500603498987

Удостоверимся, что соблюдается равенство:

In [11]:
print(s2)
print(s2_f + s2_residual)

23860.45477386935
23860.45477386935


Найдем общую дисперсию:

In [12]:
sigma2_general = s2 / (n - 1)
sigma2_general

60.10190119362557

Найдем факторную дисперсию:

In [13]:
sigma2_f = s2_f / (k - 1)
sigma2_f

4008.977085185181

Найдем остаточную дисперсию:

In [14]:
sigma2_residual = s2_residual / (n - k)
sigma2_residual

40.1075964645544

Вычислим $F_H$:

In [15]:
F_h = sigma2_f / sigma2_residual
F_h

99.95555551996654

$F_H$=99.95

Найдем значение $F_{крит}$ в таблице критических точек распределения Фишера-Снедекора для заданного уровня значимости $\alpha = 0.05$ и двух степеней свободы:

$df_{межд} = k - 1 = 3 - 1 = 2$ и $df_{внутр} = n - k = 398 - 3 = 395$.

Для данных значений $F_{крит} < 3$. Так как $F_H > F_{крит}$, отвергнута гипотеза Н0, Существенная разница существует в милях среди транспортных средств различного происхождения